pkg/services/sqlstore/component_interfaces.go

```diff
@@ -35,60 +37,64 @@ type storeDS struct {
 	ss *SQLStore
 }
 
-func (s storeDS) Get(ctx context.Context, uid string) (datasource.DataSource, error) {
+func (s storeDS) Get(ctx context.Context, uid string) (datasource.CR, error) {
 	cmd := &models.GetDataSourceQuery{
 		OrgId: 1, // Hardcode for now
 		Uid:   uid,
 	}
 
 	if err := s.ss.GetDataSource(ctx, cmd); err != nil {
-		return datasource.DataSource{}, err
+		return datasource.CR{}, err
 	}
 
 	return s.oldToNew(cmd.Result), nil
 }
 
-func (s storeDS) Insert(ctx context.Context, ds datasource.DataSource) error {
+func (s storeDS) Insert(ctx context.Context, ds datasource.CR) error {
 	cmd := &models.AddDataSourceCommand{
 		Name:              ds.Name,
-		Type:              ds.Type,
-		Access:            models.DsAccess(ds.Access),
-		Url:               ds.Url,
-		Password:          ds.Password,
-		Database:          ds.Database,
-		User:              ds.User,
-		BasicAuth:         ds.BasicAuth,
-		BasicAuthUser:     ds.BasicAuthUser,
-		BasicAuthPassword: ds.BasicAuthPassword,
-		WithCredentials:   ds.WithCredentials,
-		IsDefault:         ds.IsDefault,
-		JsonData:          simplejson.NewFromAny(ds.JsonData),
+		Type:              ds.Spec.Type,
+		Access:            models.DsAccess(ds.Spec.Access),
```

Comment:
Def need this for now, but interesting to note how referencing these old models will probably end up creating cyclic dependencies a couple steps down the line. We'll cross that bridge when we come to it!

Reply:
to do so I think that we need to provide our own implementations of querying the database rather than using the existing `SQLStore` implementations. WDYT?